### PR TITLE
introduce sync+exec watch action

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
introduce `sync+exec` watch action to run a command inside container after sync updated filesystem
TODO: e2e test

**Related issue**
closes https://github.com/docker/compose/issues/12318
requires https://github.com/compose-spec/compose-go/pull/715

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

### CODE REVIEW

**Comment by glours** [comment]
I tested with the `avatars` project and I noticed that `watch` applied a `sync` action instead of a `rebuild` one when updating the `requirement.txt` file
Behaviour with this branch:
```
> docker compose up --watch                                                                                                                                                                                                       node system 13:50:24
[+] Running 3/3
 ✔ Network avatars_default  Created                                                                                                                                                                                                                                         0.0s
 ✔ Container avatars-api-1  Created                                                                                                                                                                                                                                         0.1s
 ✔ Container avatars-web-1  Created                                                                                                                                                                                                                                         0.1s
        ⦿ Watch enabled
Attaching to api-1, web-1
web-1   | yarn run v1.22.22
api-1   |  * Serving Flask app './api/app.py' (lazy loading)
api-1   |  * Environment: development
api-1   |  * Debug mode: on
web-1   | $ vite
api-1   |  * Debugger is active!
        ⦿ Syncing service "api" after 1 changes were detected
```
Behaviour on main branch:
```
> docker compose up --watch                                                                                                                                                                                                       node system 13:47:44
[+] Running 3/3
 ✔ Network avatars_default  Created                                                                                                                                                                                                                                         0.0s
 ✔ Container avatars-web-1  Created                                                                                                                                                                                                                                         0.1s
 ✔ Container avatars-api-1  Created                                                                                                                                                                                                                                         0.1s
        ⦿ Watch enabled
Attaching to api-1, web-1
web-1   | yarn run v1.22.22
web-1   | $ vite
api-1   |  * Serving Flask app './api/app.py' (lazy loading)
api-1   |  * Environment: development
api-1   |  * Debug mode: on
api-1   |  * Debugger is active!
        ⦿ Rebuilding service "api" after changes were detected...
        ⦿ service "api" successfully built
api-1 exited with code 0
api-1 has been recreated
api-1   |  * Serving Flask app './api/app.py' (lazy loading)
api-1   |  * Environment: development
api-1   |  * Debug mode: on
api-1   |  * Debugger is active!
```

**Comment by ndeloof** [comment]
`api/requirements.txt` file event matches multiple watch rules, so the confusion :'(

File: pkg/e2e/watch_test.go [hunk 1]
```diff
@@ -289,3 +290,42 @@ func doTest(t *testing.T, svcName string) {
...
 	testComplete.Store(true)
 }
+
+func TestWatchExec(t *testing.T) {
+	cli := NewCLI(t)
+	const projectName = "test_watch_exec"
+
+	t.Cleanup(func() {
+		cli.RunDockerComposeCmd(t, "-p", projectName, "down")
+	})
```
Comment (glours) [inline pkg/e2e/watch_test.go:300]: you can now use 😉 
```suggestion
	defer c.cleanupWithDown(t, projectName)
```

**Review by glours — APPROVED** [review]
LGTM

**Pull request merged**